### PR TITLE
Remove unnecessary `defer` call

### PR DESCRIPTION
**Pull request by elland**

### CODE REVIEW

File: Library/Models/TableViewDataSource.swift [hunk 1]
```diff
@@ -100,9 +100,9 @@ class TableViewDataSource<Item: TableItem> {
                             }
                         }
                     }
-
-                    self.delegate?.dataSourceDidChangeContent()
                 }
+
+                defer { self.delegate?.dataSourceDidChangeContent() }
```
Comment (3lvis) [inline Library/Models/TableViewDataSource.swift]: Is the defer needed?

Comment (elland) [reply Library/Models/TableViewDataSource.swift]: Hmm, now I'm not so sure. 💭 What do you have in mind? Just removing defer and calling source did change directly?

Comment (3lvis) [reply Library/Models/TableViewDataSource.swift]: Yeah

Comment (elland) [reply Library/Models/TableViewDataSource.swift]: Ok. Done. Gonna try it in a bit.

**Comment by elland** [comment]
I think I broke git. 😭

**Comment by elland** [comment]
Ok, back to normal. Merging this?

**Comment by 3lvis** [comment]
Ship it!

**Pull request merged**